whitepapers/Tables.md

```diff
@@ -1,5 +1,114 @@
+| `source_doc_id` | text |  |  |  |  | Identifier given to the document by the source |
+| `ingest_config_id` | int |  |  | ✓ |  | IngestConfig used to ingest the document |
+
+#### Other indices
+* `source_doc_id`, `ingest_config_id` (unique)
+
+### RawData
+Raw data, exactly as it was given to SHARE.
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `suid_id` | int |  |  | ✓ | | SUID for this datum |
+| `data` | text |  |  |  | | The raw data itself (typically JSON or XML string) |
+| `sha256` | text | unique |  |  | | SHA-256 hash of `data` |
+| `date_seen` | datetime |  |  |  | now (every update) | The last time this exact data was harvested |
+| `date_harvested` | datetime |  |  |  | now (on insert) | The first time this exact data was harvested |
+
+## Ingest Configuration
+
+### IngestConfig
+Describes one way to harvest metadata from a Source, and how to transform the result.
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `source_id` | int |  |  | ✓ | | Source to harvest from |
+| `base_url` | text |  |  |  | | URL of the API or endpoint where the metadata is available |
+| `earliest_date` | date |  | ✓ |  | | Earliest date with available data |
+| `rate_limit_allowance` | int |  |  |  | 5 | Number of requests allowed every `rate_limit_period` seconds |
+| `rate_limit_period` | int |  |  |  | 1 | Number of seconds for every `rate_limit_allowance` requests |
+| `harvester_id` | int |  |  | ✓ | | Harvester to use |
+| `harvester_kwargs` | jsonb |  | ✓ |  | | JSON object passed to the harvester as kwargs |
+| `transformer_id` | int |  |  | ✓ | | Transformer to use |
+| `transformer_kwargs` | jsonb |  | ✓ |  | | JSON object passed to the transformer as kwargs, along with the harvested raw data |
+| `disabled` | bool |  |  |  | False | True if this ingest config should not be run automatically |
+
+### Source
+A Source is a place metadata comes from.
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `name` | text | unique |  |  | | Short name |
+| `long_title` | text | unique |  |  | | Full, human-friendly name |
+| `home_page` | text |  | ✓ |  | | URL |
+| `icon` | image |  | ✓ |  | | Recognizable icon for the source |
+| `user_id` | int |  |  | ✓ | | User with permission to submit data as this source (TODO: replace with django permissions stuff) |
+
+### Harvester
+Each row corresponds to a Harvester implementation in python. (TODO: describe those somewhere)
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `key` | text | unique |  |  | | Key that can be used to get the corresponding Harvester subclass |
+| `date_created` | datetime |  |  |  | now (on insert) | |
```

Comment:
If we implement this with django's `auto_now_add`, `NOW()` isn't used and there's no actual default in postgres. Does that matter here?

Or in another case, `Source.icon` is technically a text field, with a value which corresponds to where the image is stored, because that's how django's `ImageField` works. That seems like irrelevant/extraneous detail for a white paper, and all that actually matters is that each `Source` has an (optional) image named `icon`.

How much should this file be considered exact specs for SQL tables, vs. a more abstract view of the objects and their attributes, with implementation details left up to the reader? I feel like we need to find a balance where these papers accurately describe a system, and our code is but one possible implementation of that system. But I'm not quite sure where that balance lies.